### PR TITLE
Fixed invalid additionalInfo validation in enroll offer modal

### LINKED ISSUE #2867: (SP: 1) Fix invalid additionalInfo in offer request is sent to the server
Validate additionalInfo in Enroll offer modal window to be 30-1000 characters

image

Modal window "Enroll offer":
image

1. Send a request withoud additionalInfo (empty string)
2. Recieve an error in logs
```
{
	level: 'error',
	message: 'Cooperation validation failed: additionalInfo: Additional info cannot be shorter than 30 symbol.',
}
```

### PR DESCRIPTION
**Pull request by VKormylo**
Added validation for additional info to be at least 30 symbols or empty in enroll offer modal.
Validation looks like this.
image
If the field is empty, there is no validation.
image

### CODE REVIEW

File: src/components/app-text-area/AppTextArea.tsx
Comment (markgol777) [inline src/components/app-text-area/AppTextArea.tsx:1]: suggestion: Maybe put the ternary into a separate variable for better code reading

**Review by markgol777 — APPROVED** [review]

File: src/containers/offer-details/enroll-offer/EnrollOffer.styles.ts
Comment (markgol777) [inline src/containers/offer-details/enroll-offer/EnrollOffer.styles.ts:1]: It's not a good practise to use negative margins

**Review by markgol777 — CHANGES_REQUESTED** [review]

File: src/components/app-text-area/AppTextArea.tsx
Comment (VKormylo) [reply src/components/app-text-area/AppTextArea.tsx:1]: Yes, I agree.

File: src/containers/offer-details/enroll-offer/EnrollOffer.styles.ts
Comment (VKormylo) [reply src/containers/offer-details/enroll-offer/EnrollOffer.styles.ts:1]: I'm going to fix that.

Comment (VKormylo) [reply src/containers/offer-details/enroll-offer/EnrollOffer.styles.ts:1]: Fixed.

**Review by markgol777 — APPROVED** [review]

**Review by luiqor — COMMENTED** [review]
Additional info cannot be empty "" (0 characters), as it causes a server error. It must either be not included to request or contain more than 30 characters. Please update the logic to ensure that additionalInfo is not sent if the input field is empty string.

image
image

**Comment by VKormylo** [comment]
Fixed incorrect empty additionalInfo in request

**Review by luiqor — APPROVED** [review]

**Review by SofiiaYevush — APPROVED** [review]

**Pull request merged**